specification/snapshot/_types/SnapshotRepository.ts

```diff
@@ -17,22 +17,126 @@
  * under the License.
  */
 
-import { Uuid } from '@_types/common'
+import { ByteSize, Uuid } from '@_types/common'
 import { integer } from '@_types/Numeric'
+import { Duration } from '@_types/Time'
 
-export class Repository {
-  type: string
+export type Repository =
+  | AzureRepository
+  | GcsRepository
+  | S3Repository
+  | SharedFileSystemRepository
+  | ReadOnlyUrlRepository
+  | SourceOnlyRepository
+
+export class AzureRepository {
+  type: 'azure'
+  uuid?: Uuid
```

Comment:
Move `uuid` to a common `RepositoryBase` so that we can access the id of a repository without having to downcast it to its actual type.

Reply:
Fixed in f9550144a5cf45ba5389786ea73132c6859db39b.